frontend/src/components/resourceMap/GraphView.tsx

```diff
@@ -0,0 +1,390 @@
+                />
+
+                <Chip
+                  label={t('Status: Error or Warning')}
+                  color={hasErrorsFilter ? 'primary' : undefined}
+                  variant={hasErrorsFilter ? 'filled' : 'outlined'}
+                  icon={hasErrorsFilter ? <Icon icon="mdi:check" /> : undefined}
+                  onClick={() => setHasErrorsFilter(!hasErrorsFilter)}
+                  sx={{
+                    lineHeight: '1',
+                  }}
+                />
+
+                {graphSize < 50 && (
+                  <Chip
+                    label={t('Expand All')}
+                    color={expandAll ? 'primary' : undefined}
+                    variant={expandAll ? 'filled' : 'outlined'}
+                    icon={expandAll ? <Icon icon="mdi:check" /> : undefined}
+                    onClick={() => setExpandAll(it => !it)}
+                    sx={{
+                      lineHeight: '1',
+                    }}
+                  />
+                )}
+              </Box>
+            </Box>
+            <div style={{ flexGrow: 1 }}>
+              <GraphRenderer
+                nodes={layoutedGraph.nodes}
+                edges={layoutedGraph.edges}
+                onMoveStart={e => {
+                  if (e === null) return;
+                  viewportMovedRef.current = true;
+                }}
+                onBackgroundClick={() => {
+                  if (selectedGroup && selectedNodeId === selectedGroup.id) {
+                    const parent = getParentNode(fullGraph, selectedGroup.id);
+                    if (parent) {
+                      setSelectedNodeId(parent.id);
+                    }
+                  } else if (selectedGroup) {
+                    setSelectedNodeId(selectedGroup.id);
+                  }
+                }}
+                onNodeClick={useCallback((event: any, node: any) => {
+                  if (isGroup(node)) {
+                    setSelectedNodeId(node.id);
+                  }
+                }, [])}
+              >
+                <Panel position="top-left">
+                  {selectedGroup && (
+                    <SelectionBreadcrumbs
+                      graph={fullGraph}
+                      selectedNodeId={selectedNodeId}
+                      onNodeClick={id => setSelectedNodeId(id)}
+                    />
+                  )}
+                </Panel>
+              </GraphRenderer>
+            </div>
+          </Box>
+        </CustomThemeProvider>
+        {selectedNode && (
+          <GraphNodeDetails
+            node={selectedNode}
+            close={() => {
+              if (selectedGroup) {
+                setSelectedNodeId(selectedGroup.id);
+              } else {
+                setSelectedNodeId(defaultNodeSelection);
+              }
+            }}
+          />
+        )}
+      </Box>
+    </GraphViewContext.Provider>
+  );
+}
+
+function CustomThemeProvider({ children }: { children: ReactNode }) {
```

Comment:
Could this be integrated into the main theme?

Reply:
Our current app theme is something I want to work on separately

Reply:
Yes, this code seems strange. But let's skip this to avoid delay and argument.